build_runner_core/lib/src/package_graph/apply_builders.dart

```diff
@@ -268,17 +268,18 @@ Future<List<BuildPhase>> createBuildPhases(
         (globalOptions[key] ?? BuilderOptions.empty).overrideWith(overrides);
   }
 
-  final cycles = stronglyConnectedComponents<String, TargetNode>(
+  final cycles = stronglyConnectedComponents<TargetNode>(
       targetGraph.allModules.values,
-      (node) => node.target.key,
       (node) => node.target.dependencies?.map((key) {
             if (!targetGraph.allModules.containsKey(key)) {
               _logger.severe('${node.target.key} declares a dependency on $key '
                   'but it does not exist');
               throw CannotBuildException();
             }
             return targetGraph.allModules[key];
-          })?.where((n) => n != null));
+          })?.where((n) => n != null),
+      equals: (a, b) => a.target.key == b.target.key,
+      hashCode: (node) => node.target.key.hashCode);
```

Comment:
nit: try a trailing common at the end of the param list, I think it makes it easier to read

Reply:
It has almost zero impact here... I don't think it's worth the extra line

Reply:
Bobs fears have now been fully realized :sob: :sob: :sob: